apps/aecore/src/aecore_app.erl

```diff
@@ -49,10 +36,13 @@ stop(_State) ->
     ok.
 
 set_app_ctrl_mode() ->
-    Mode = case ok(aeu_env:find_config([<<"system">>, <<"maintenance_mode">>],
-                                       [user_config, schema_default])) of
-               true  -> maintenance;
-               false -> normal
+    MMode = ok(aeu_env:find_config([<<"system">>, <<"maintenance_mode">>],
+                                   [user_config, schema_default])),
+    DevMode = is_dev_mode_consensus(),
+    Mode = case {MMode, DevMode} of
+               {true , _}     -> maintenance;
+               {false, true}  -> dev_mode;
```

Comment:
I'm a bit confused to see this feature as special "mode"

Reply:
The 'on_demand' consensus protocol is special: once activated, it cannot be deactivated. Also, e.g. sync needs to be disabled, since a bunch of integrity checks are disabled. It seems reasonable that the node is aware of this.